### PR TITLE
Added Profile Page

### LINKED ISSUE #121: [New feature] User profile page
Create a profile icon on the top of naviagtion bar 
When user clicks on that, open up a new page `user-profile`
Show all the user details in that page.

### PR DESCRIPTION
**Pull request by syedjahangirpeeran**
Fixes #121 .

### CODE REVIEW

**Comment by Swati4star** [comment]
Add a screenshot also.

File: Android/Android.iml [hunk 1]
```diff
@@ -13,7 +13,7 @@
     <content url="file://$MODULE_DIR$">
       <excludeFolder url="file://$MODULE_DIR$/.gradle" />
     </content>
-    <orderEntry type="jdk" jdkName="1.8" jdkType="JavaSDK" />
+    <orderEntry type="inheritedJdk" />
```
Comment (Swati4star) [inline Android/Android.iml]: Why this change?

File: Android/app/src/main/AndroidManifest.xml [hunk 2]
```diff
@@ -119,8 +121,11 @@
             android:grantUriPermissions="true">
             <meta-data
                 android:name="android.support.FILE_PROVIDER_PATHS"
-                android:resource="@xml/provider_paths"/>
+                android:resource="@xml/provider_paths" />
         </provider>
+
+        <activity android:name=".ProfileActivity"
+            android:parentActivityName=".login.LoginActivity"/>
```
Comment (Swati4star) [inline Android/app/src/main/AndroidManifest.xml]: Why are you making other refactoring changes. You just need to add this ProfileActivity

File: Android/app/src/main/res/values/strings.xml [hunk 3]
```diff
@@ -162,5 +162,9 @@
     <string name="signout_message">Are you sure want to sign out?</string>
     <string name="positive_button">OK</string>
     <string name="negative_button">CANCEL</string>
+    <string name="member_since" />
```
Comment (Swati4star) [inline Android/app/src/main/res/values/strings.xml]: You have not added values to the strings

File: Android/app/src/main/res/layout/activity_profile.xml [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+        android:layout_alignParentBottom="true"
+        android:layout_centerHorizontal="true"
+        android:layout_marginBottom="191dp"
+        android:paddingTop="6sp"
+        android:text="Default Name"
+        android:textColor="#333434"
+        android:textSize="18sp" />
+
+    <ImageView
+        android:id="@+id/imageView2"
```
Comment (Swati4star) [inline Android/app/src/main/res/layout/activity_profile.xml]: don't make id name as `imageview`
name them what they actually are

File: Android/app/src/main/res/layout/activity_profile.xml [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+<RelativeLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    android:background="@color/colorPrimary"
+    tools:context=".ProfileActivity">
+
+    <TextView
+        android:id="@+id/display_exp"
```
Comment (Swati4star) [inline Android/app/src/main/res/layout/activity_profile.xml]: `display_joining_date`

File: Android/app/src/main/res/layout/activity_profile.xml [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+        android:layout_height="46dp"
+        android:layout_above="@+id/change_image_btn"
+        android:layout_centerHorizontal="true"
+        android:paddingTop="6sp"
+        android:text="Member Since"
+        android:textColor="#333434"
+        android:textSize="18sp" />
+
+    <TextView
+        android:id="@+id/display_email2"
```
Comment (Swati4star) [inline Android/app/src/main/res/layout/activity_profile.xml]: `display_email`

File: Android/app/src/main/res/layout/activity_profile.xml [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+        android:id="@+id/settings_image"
+        android:layout_width="220dp"
+        android:layout_height="220dp"
+        android:layout_alignParentTop="true"
+        android:layout_centerHorizontal="true"
+        android:layout_marginTop="24dp"
+        android:src="@drawable/default_user_icon" />
+
+    <Button
+        android:id="@+id/change_image_btn"
```
Comment (Swati4star) [inline Android/app/src/main/res/layout/activity_profile.xml]: what is this button for?

File: Android/app/src/main/java/io/github/project_travel_mate/ProfileActivity.java [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+    TextView email;
+    @BindView(R.id.display_exp)
+    TextView exp;
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_profile);
+        ButterKnife.bind(this);
+        SharedPreferences preferences = PreferenceManager.getDefaultSharedPreferences(this);
+        String Email = preferences.getString(USER_EMAIL, "Email ID");
```
Comment (Swati4star) [inline Android/app/src/main/java/io/github/project_travel_mate/ProfileActivity.java]: change `Email` to `email`

File: Android/app/src/main/java/io/github/project_travel_mate/ProfileActivity.java [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+        Log.d("test", Email);
+        email.setText(Email);
+        String Name = preferences.getString("firstname", "Display name");
```
Comment (Swati4star) [inline Android/app/src/main/java/io/github/project_travel_mate/ProfileActivity.java]: `Name` to `name`

File: Android/app/src/main/java/io/github/project_travel_mate/ProfileActivity.java [hunk 10]
```diff
@@ -0,0 +1,54 @@
...
+    @BindView(R.id.display_exp)
+    TextView exp;
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_profile);
+        ButterKnife.bind(this);
+        SharedPreferences preferences = PreferenceManager.getDefaultSharedPreferences(this);
+        String Email = preferences.getString(USER_EMAIL, "Email ID");
+        Log.d("test", Email);
```
Comment (Swati4star) [inline Android/app/src/main/java/io/github/project_travel_mate/ProfileActivity.java]: Remove logs

File: Android/app/src/main/java/io/github/project_travel_mate/ProfileActivity.java (hunk 9 above)
Comment (Swati4star) [reply Android/app/src/main/java/io/github/project_travel_mate/ProfileActivity.java]: Add `firstname` to constants file & use that here

File: Android/Android.iml (hunk 1 above)
Comment (syedjahangirpeeran) [reply Android/Android.iml]: The project is Gradle-based, So Android Studio changes it automatically a/c to to it's version. InheritedJDk is default and runs on all computers. Nothing to do with app functionality.

File: Android/app/src/main/AndroidManifest.xml (hunk 2 above)
Comment (syedjahangirpeeran) [reply Android/app/src/main/AndroidManifest.xml]: Sorry forgot to remove

File: Android/app/src/main/res/layout/activity_profile.xml (hunk 7 above)
Comment (syedjahangirpeeran) [reply Android/app/src/main/res/layout/activity_profile.xml]: To change profile pic. Shall I remove it?

**Comment by syedjahangirpeeran** [comment]
Changes Made.
screenshot_20180625-141319

**Comment by syedjahangirpeeran** [comment]
@Swati4star Changes Made.

**Comment by Swati4star** [comment]
The display name is not aligned properly with the icon. And the left margin of other fields also doesn't align properly.

**Comment by Swati4star** [comment]
Also remove the refactoring changes for manifest.xml
Add only the required changes.

**Comment by Swati4star** [comment]
Post the new screenshot of profile page after making the changes.

File: Android/app/src/main/java/io/github/project_travel_mate/ProfileActivity.java [hunk 11]
```diff
@@ -0,0 +1,53 @@
...
+
+import butterknife.BindView;
+import butterknife.ButterKnife;
+
+import static utils.Constants.FIRST_NAME;
+import static utils.Constants.USER_EMAIL;
+
+public class ProfileActivity extends AppCompatActivity {
+    @BindView(R.id.display_name)
+    TextView display_name;
```
Comment (Swati4star) [inline Android/app/src/main/java/io/github/project_travel_mate/ProfileActivity.java]: The naming convention for variable names is Camel case. Change these to 
```
displayName
emailId
joiningDate
```

File: Android/app/src/main/java/io/github/project_travel_mate/ProfileActivity.java [hunk 12]
```diff
@@ -0,0 +1,53 @@
...
+    TextView email_id;
+    @BindView(R.id.display_joining_date)
+    TextView joining_date;
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_profile);
+        ButterKnife.bind(this);
+        SharedPreferences preferences = PreferenceManager.getDefaultSharedPreferences(this);
+        String email = preferences.getString(USER_EMAIL, "Email ID");
```
Comment (Swati4star) [inline Android/app/src/main/java/io/github/project_travel_mate/ProfileActivity.java]: Change default value from `"Email ID" to "--"`

File: Android/app/src/main/java/io/github/project_travel_mate/ProfileActivity.java [hunk 13]
```diff
@@ -0,0 +1,53 @@
...
+        email_id.setText(email);
+        String name = preferences.getString(FIRST_NAME, "Display name");
```
Comment (Swati4star) [inline Android/app/src/main/java/io/github/project_travel_mate/ProfileActivity.java]: Do the same thing here also.

File: Android/app/src/main/res/menu/activity_main_drawer.xml [hunk 14]
```diff
@@ -2,6 +2,10 @@
 <menu xmlns:android="http://schemas.android.com/apk/res/android">
 
     <group android:checkableBehavior="single">
+        <item
+            android:id="@+id/nav_profile"
+            android:icon="@drawable/icon_profile"
+            android:title="Profile" />
```
Comment (Swati4star) [inline Android/app/src/main/res/menu/activity_main_drawer.xml]: Don't add constant string here. Move `Profile` to strings.xml & then, use it here.

**Comment by Swati4star** [comment]
Also post the screenshot of navigation drawer. I want to see how it looks after adding this option over there.

**Comment by brijeshshah13** [comment]
@Swati4star I think the background color should be white. Currently the whole screen is blue which is looking pretty poor.

**Comment by Swati4star** [comment]
@syedjahangirpeeran Also post a screenshot of profile page with white UI.

**Comment by syedjahangirpeeran** [comment]
`<RelativeLayout xmlns:android="http://schemas.android.com/apk/res/android"
    xmlns:app="http://schemas.android.com/apk/res-auto"
    xmlns:tools="http://schemas.android.com/tools"
    android:background="@color/white"
    android:layout_width="match_parent"
    android:layout_height="match_parent"
    tools:context=".ProfileActivity">`
I even tried `android:background="#ffffff"` but the background color is changing only in the preview and not in app. Any thoughts why?

**Comment by syedjahangirpeeran** [comment]
screenshot_1529988864

**Comment by Swati4star** [comment]
We don't have different layout files for v4. You can remove that one.

**Comment by Swati4star** [comment]
Don't add a new option in the navigation bar. Change the icon (the person icon which is there on the top of navigation bar) to profile icon & on clicking that, the profile should open up. Adding it as separate option on navigation bar doesn't look good.

File: Android/app/src/main/java/io/github/project_travel_mate/MainActivity.java [hunk 15]
```diff
@@ -87,6 +87,12 @@ public boolean onNavigationItemSelected(@NonNull MenuItem item) {
         FragmentManager fragmentManager = getSupportFragmentManager();
 
         switch (id) {
+            case R.id.nav_profile:
+                Intent in = new Intent(MainActivity.this, ProfileActivity.class);
+                startActivity(in);
+                finish();
```
Comment (Swati4star) [inline Android/app/src/main/java/io/github/project_travel_mate/MainActivity.java]: Don't finish the current activity. Let it stay in the back.

File: Android/app/src/main/java/io/github/project_travel_mate/ProfileActivity.java [hunk 16]
```diff
@@ -0,0 +1,52 @@
...
+        Objects.requireNonNull(getSupportActionBar()).setHomeButtonEnabled(true);
+        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
+    }
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
+        switch (item.getItemId()) {
+            case android.R.id.home:
+                // app icon in action bar clicked; go home
+                Intent intent = new Intent(this, MainActivity.class);
+                intent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP);
```
Comment (Swati4star) [inline Android/app/src/main/java/io/github/project_travel_mate/ProfileActivity.java]: You won't need to start new activity here now. You can just finish this one only.

**Comment by syedjahangirpeeran** [comment]
screenshot_20180626-104840
screenshot_20180626-104835

File: Android/app/src/main/res/values/strings.xml [hunk 17]
```diff
@@ -162,5 +162,15 @@
     <string name="signout_message">Are you sure want to sign out?</string>
     <string name="positive_button">OK</string>
     <string name="negative_button">CANCEL</string>
+    <string name="display_email" >Display Email</string>
+    <string name="display_name" >Display Name</string>
+    <string name="default_name">Default Name</string>
+    <string name="email">Email</string>
+    <string name="member_from">Member From</string>
```
Comment (Swati4star) [inline Android/app/src/main/res/values/strings.xml]: Make `F` small

File: Android/app/src/main/java/io/github/project_travel_mate/MainActivity.java [hunk 18]
```diff
@@ -171,4 +172,10 @@ void getRuntimePermissions() {
             }
         }
     }
+
+    public void onClickProfile(View view) {
+        Intent in = new Intent(MainActivity.this, ProfileActivity.class);
+        startActivity(in);
+        finish();
```
Comment (Swati4star) [inline Android/app/src/main/java/io/github/project_travel_mate/MainActivity.java]: Don't finish main activity.

**Comment by Swati4star** [comment]
The page with white background looks better. @syedjahangirpeeran 
Thanks for the suggestion, @brijeshshah13 👍

**Comment by syedjahangirpeeran** [comment]
@brijeshshah13  it was due to alpha value. I increased it now!

**Comment by brijeshshah13** [comment]
@syedjahangirpeeran Can you please post the updated screenshot for better reference?

**Comment by syedjahangirpeeran** [comment]
@Swati4star  mam, please go through the changes.

**Comment by Swati4star** [comment]
Please post the final screenshots also. @syedjahangirpeeran

**Comment by Swati4star** [comment]
The code changes look good now.

**Comment by syedjahangirpeeran** [comment]
screenshot_20180626-152421

**Comment by syedjahangirpeeran** [comment]
screenshot_20180626-152429

**Comment by Swati4star** [comment]
Make the icon white in navigation drawer.
You can use this one https://material.io/tools/icons/?icon=account_circle&style=baseline in white color

**Comment by syedjahangirpeeran** [comment]
screenshot_20180626-154618

**Comment by Swati4star** [comment]
Just squash the changes into one commit & then, I'll merge this PR.

**Comment by syedjahangirpeeran** [comment]
Done!

**Comment by Swati4star** [comment]
Can you reword the commit? It doesn't give any information about the issue being solved.

**Comment by Swati4star** [comment]
I'll merge it.

**Comment by Swati4star** [comment]
Merged in c3228b7a1f51cf955ebc45debcc1c25be5b588c4 👍
